File: solutions/_145_binary_tree_postorder_traversal/Solution.java

package _145_binary_tree_postorder_traversal;

import java.util.*;

class Solution {
    public List<Integer> postorderTraversal(TreeNode root) {
        List<Integer> res = new LinkedList<Integer>();
        if (root == null) {
            return res;
        }
        Stack<TreeNode> stack = new Stack<>();
        stack.push(root);

        while (!stack.isEmpty()) {
            var node = stack.pop();
            res.add(0, node.val);
            if (node.left != null) {
                stack.push(node.left);
            }
            if (node.right != null) {
                stack.push(node.right);
            }
        }

        return res;
    }
}

/*
 * https://leetcode.com/problems/binary-tree-postorder-traversal/
 */

File: solutions/_145_binary_tree_postorder_traversal/solution.js
function TreeNode(val, left, right) {
  this.val = val === undefined ? 0 : val;
  this.left = left === undefined ? null : left;
  this.right = right === undefined ? null : right;
}

/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var postorderTraversal = function (root) {
  if (!root) return [];
  const stack = [root];
  const res = [];

  while (stack.length) {
    const node = stack.pop();
    res.unshift(node.val);
    if (node.left) {
      stack.push(node.left);
    }
    if (node.right) {
      stack.push(node.right);
    }
  }

  return res;
};

module.exports = { postorderTraversal, TreeNode };

/*
https://leetcode.com/problems/binary-tree-postorder-traversal/
*/
